BDArmory/WeaponMounts/ModuleCustomTurret.cs

```diff
@@ -149,28 +215,62 @@ public void AimInDirection(Vector3 targetDirection)
             }
             if (!fullRotation && Mathf.Abs(currentYaw - targetYawAngle) >= 180)
             {
+                //if (float.IsNaN(currentYaw)) return;
                 targetYawAngle = currentYaw - (Math.Sign(currentYaw) * 179);
             }
             if (Servo)
             {
                 Servo.targetAngle = targetYawAngle;
                 if (Servo.inverted) Servo.targetAngle *= -1;
-                Debug.Log($"[CUSTOMTURRET] CurrYaw: {currentYaw}; YawError: {yawError}; Servo Yaw: {Servo.currentAngle}; Servo target Angle {Servo.targetAngle}");
+                //Debug.Log($"[BDArmory.ModuleCustomTurret] CurrYaw: {currentYaw}; YawError: {yawError}; Servo target Angle {Servo.targetAngle}; Servo Yaw: {Servo.currentAngle}");
             }
             if (Hinge)
             {
                 float pitchError = (float)VectorUtils.SignedAngleDP(pitchComponent, yawNormal, Hinge.mainAxis == "X" ? pitchTransform.right : pitchTransform.forward) - (float)VectorUtils.SignedAngleDP(referenceTransform.forward, yawNormal, Hinge.mainAxis == "X" ? pitchTransform.right : pitchTransform.forward);
                 //float currentPitch = Hinge ? Hinge.currentAngle : Servo ? 0 : 0;
-                float currentPitch = Hinge ? VectorUtils.SignedAngleDP(Hinge.mainAxis == "X" ? bottomTransform.up : bottomTransform.right, referenceTransform.forward, bottomTransform.forward) : Servo ? 0 : 0;
+                float currentPitch = Hinge ? -VectorUtils.SignedAngleDP(baseForward, Hinge.mainAxis == "X" ? pitchTransform.up : Hinge.mainAxis == "Z" ? pitchTransform.forward : -pitchTransform.right,
```

Comment:
This is unstable when on-target due to `toDirection` and `referenceRight` being orthogonal. `referenceRight` should preferably be orthogonal to `fromDirection`. I suspect this would be easiest by adding a `baseRight` too.